### PR TITLE
config_tools: add hybrid, hybrid_rt, logical partition scenario on ad…

### PR DESCRIPTION
**Pull request by szhen11**
…l board

add hybrid, hybrid_rt, logical_partition scenario and add launch
config xml on adl-s-crb board, add launch config xml on adl-rvp board.

Tracked-On: #5929
Signed-off-by: Shuang Zheng <shuang.zheng@intel.com>

### CODE REVIEW

File: misc/config_tools/data/adl-s-crb/logical_partition.xml [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+      <size>0x20000000</size>
+      <start_hpa2>0x0</start_hpa2>
+      <size_hpa2>0x0</size_hpa2>
+    </memory>
+    <os_config>
+      <name>YOCTO</name>
+      <kern_type>KERNEL_BZIMAGE</kern_type>
+      <kern_mod>Linux_bzImage</kern_mod>
+      <ramdisk_mod></ramdisk_mod>
+      <bootargs>        rw rootwait root=/dev/sda3 console=ttyS0 noxsave nohpet no_timer_check ignore_loglevel log_buf_len=16M consoleblank=0 tsc=reliable reboot=acpi
```
Comment (gvancuts) [inline misc/config_tools/data/adl-s-crb/logical_partition.xml:94]: Same question about the leading spaces

File: misc/config_tools/data/adl-s-crb/logical_partition.xml [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+      <size>0x20000000</size>
+      <start_hpa2>0x0</start_hpa2>
+      <size_hpa2>0x0</size_hpa2>
+    </memory>
+    <os_config>
+      <name>YOCTO</name>
+      <kern_type>KERNEL_BZIMAGE</kern_type>
+      <kern_mod>Linux_bzImage</kern_mod>
+      <ramdisk_mod></ramdisk_mod>
+      <bootargs>        rw rootwait root=/dev/sda3 console=ttyS0 noxsave nohpet no_timer_check ignore_loglevel log_buf_len=16M
```
Comment (gvancuts) [inline misc/config_tools/data/adl-s-crb/logical_partition.xml:166]: Leading spaces?

**Review by gvancuts — CHANGES_REQUESTED** [review]
A couple of issues with this PR:
1. It adds a number of files for a new board (`adl-s-crb`) but no such `adl-s-crb.xml` that describes the board itself
2. We are adding yet another board to the list, we should revisit our strategy around these here (and involved the UX/DX in the conversation)

File: misc/config_tools/data/adl-rvp/industry_launch_2uos.xml [hunk 3]
```diff
@@ -0,0 +1,89 @@
+<acrn-config board="adl-s-crb" scenario="industry" uos_launcher="2">
```
[the remark this answers is not in the document]
Comment (szhen11) [reply misc/config_tools/data/adl-rvp/industry_launch_2uos.xml:1]: sorry for this mistake, will update the board attribute value to adl-rvp.

File: misc/config_tools/data/adl-s-crb/hybrid_rt.xml [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+      <base>INVALID_PCI_BASE</base>
+      <target_vm_id>0</target_vm_id>
+      <target_uart_id>1</target_uart_id>
+    </communication_vuart>
+    <pci_devs>
+      <pci_dev></pci_dev>
+    </pci_devs>
+    <board_private>
+      <rootfs>/dev/nvme0n1p3</rootfs>
+      <bootargs>            rw rootwait console=tty0 consoleblank=0 no_timer_check quiet loglevel=3
```
[the remark this answers is not in the document]
Comment (szhen11) [reply misc/config_tools/data/adl-s-crb/hybrid_rt.xml:167]: will remove the leading space then.

File: misc/config_tools/data/adl-s-crb/logical_partition.xml (hunk 1 above)
Comment (szhen11) [reply misc/config_tools/data/adl-s-crb/logical_partition.xml:94]: will remove the leading space then.

File: misc/config_tools/data/adl-s-crb/logical_partition.xml (hunk 2 above)
Comment (szhen11) [reply misc/config_tools/data/adl-s-crb/logical_partition.xml:166]: will remove the leading space then.

File: misc/config_tools/data/adl-rvp/industry_launch_2uos.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (gvancuts) [reply misc/config_tools/data/adl-rvp/industry_launch_2uos.xml]: > sorry for this mistake, will update the board attribute value to adl-rvp.

Thanks. How are we testing these files?

> and for adl-s-crb, it is another kind of board for alder lake-s platform, we put the board xml(adl-s-crb.xml) and scenario xmls into misc/config_tools/data/adl-s-crb/

I do not see this `adl-s-crb.xml` file in this PR, have I missed it?

**Comment by gvancuts** [comment]
> 
> 
> …l board
> 
> add hybrid, hybrid_rt, logical_partition scenario and add launch
> config xml on adl-s-crb board, add launch config xml on adl-rvp board.

Can you polish up your commit message? E.g.:

```
config_tools: add 'adl-s-crb' board XMLs and launch script ('adl-rvp')

This commit adds the following data to the ACRN Configuration Tools:
* Board and scenario XMLs for a new board called 'adl-s-crb.xml'
* Add a new launch script for the 'adl-rvp' board
```

**Comment by gvancuts** [comment]
I have lost track a bit of what gets generated by the tool and what we need to add manually as input for the tool... but aren't these XML files supposed to be generated by the ACRN Configuration Tool after one has fed it the `adl-s-crb.xml` board info and configured the scenarios? It does not seem like this was the process that was used here to create those. Can someone confirm this? @junjiemao1 @NanlinXie

File: misc/config_tools/data/adl-rvp/industry_launch_2uos.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (NanlinXie) [reply misc/config_tools/data/adl-rvp/industry_launch_2uos.xml]: We have already launched the discussion for these board files location, let's keep this in v2.5 release.

**Comment by NanlinXie** [comment]
> I have lost track a bit of what gets generated by the tool and what we need to add manually as input for the tool... but aren't these XML files supposed to be generated by the ACRN Configuration Tool after one has fed it the `adl-s-crb.xml` board info and configured the scenarios? It does not seem like this was the process that was used here to create those. Can someone confirm this? @junjiemao1 @NanlinXie

Usually we will upload the scenario files that our validation major use.

**Review by NanlinXie — APPROVED** [review]

**Pull request merged**